parachain/runtime/interlay/src/xcm_config.rs

```diff
@@ -155,6 +162,32 @@ impl FixedConversionRateProvider for MyFixedConversionRateProvider {
     }
 }
 
+/// A call filter for the XCM Transact instruction. This is a temporary measure until we properly
+/// account for proof size weights.
+///
+/// Calls that are allowed through this filter must:
+/// 1. Have a fixed weight;
+/// 2. Cannot lead to another call being made;
+/// 3. Have a defined proof size weight, e.g. no unbounded vecs in call parameters.
+pub struct SafeCallFilter;
+impl Contains<RuntimeCall> for SafeCallFilter {
+    fn contains(call: &RuntimeCall) -> bool {
+        // we need to filter all calls that can recurse. We're being a bit overly conservative here
+        // by completly blocking the pallets below rather than filter per specific call.
+        match call {
+            RuntimeCall::Sudo(..) | RuntimeCall::Proxy(..) | RuntimeCall::Multisig(..) | RuntimeCall::Utility(..) => {
+                // these calls can recurse - disallow
+                false
+            }
+            RuntimeCall::Issue(..) | RuntimeCall::Replace(..) | RuntimeCall::Redeem(..) | RuntimeCall::BTCRelay(..) => {
+                // disallow anything to do with btc transactions since btc tx may be unbounded
+                false
+            }
```

Comment:
We should remember to change this once the weight refactoring is complete.

Reply:
It still wouldn't meet requirement #3 as I understand it (`Have a defined proof size weight, e.g. no unbounded vecs in call parameters.`)